EIPS/eip-5008.md

```diff
@@ -0,0 +1,174 @@
+---
+eip: 5008
+title: ERC-721 Nonce and Metadata Update Extension
+description: Add a `nonce` function an a `MetadataUpdate` event to ERC-721.
+author: Anders (@0xanders), Lance (@LanceSnow), Shrug <shrug@emojidao.org>
+discussions-to: https://ethereum-magicians.org/t/eip5008-erc-721-nonce-and-metadata-update-extension/8925
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-04-10
+requires: 165, 721
+---
+
+## Abstract
+
+This standard is an extension of [ERC-721](./eip-721.md). It proposes adding a `nonce` function and `MetadataUpdate` event to ERC-721 tokens.
+
+## Motivation
+
+Some orders of NFT marketplace has been attacked and the NFTs have been sold in a lower price than market floor price. One reason is that users transfer NFT to another wallet and then, after a certain period of time, transfer it back to the original wallet, and the order becomes valid again.
+
+This EIP proposes adding a `nonce` property to ERC-721 tokens, and the `nonce` will be changed when transfer. If a `nonce` is added to an order, the order can be checked to avoid attacks.
+
+Many [ERC-721](./eip-721.md) contracts emit their custom event when metadata changed. It is easy to update metadata of one NFT by specific event, but it is difficult for third-party platforms such as NFT marketplace to update metadata of many NFTs based on custom events.
+
+Having a standard `MetadataUpdate` event will make it easy for third-party platforms to timely update metadata of many NFTs.
+
+## Specification
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY" and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+```solidity
+interface IERC5008  {
+    /// @dev This event emits when the metadata of a token is changed.
+    /// So that the third-party platforms such as NFT market could
+    /// timely update the images and related attributes of the NFT
+    event MetadataUpdate(uint256 tokenId);
```

Comment:
It almost feels like `MetadataUpdate` should be in a separate EIP from `nonce`. They _seem_ completely independent?

Reply:
> 

Yes, `MetadataUpdate`  in a separate EIP  is better. I reopened  https://github.com/ethereum/EIPs/pull/4906